### DISCUSSION TITLE
Could not find org.apache.cordova:framework:7.0.0

### DISCUSSION THREAD

**Discussion by joshstrange** [ionic-team/capacitor#6672] [General]
## Bug Report

You can no longer install any version of the cordova framework <10

### Capacitor Version
```
💊   Capacitor Doctor  💊

Latest Dependencies:

  @capacitor/cli: 5.0.5
  @capacitor/core: 5.0.5
  @capacitor/android: 5.0.5
  @capacitor/electron: 5.0.5
  @capacitor/ios: 5.0.5

Installed Dependencies:

  @capacitor/cli 2.5.0
  @capacitor/core 2.5.0
  @capacitor/ios 2.5.0
  @capacitor/android 2.5.0
  @capacitor/electron not installed

[success] Android looking great! 👌
  Found 2 Capacitor plugins for ios:
    cordova-open-native-settings (1.5.5)
    cordova-plugin-safariviewcontroller (2.0.0)
[success] iOS looking great! 👌
```

### Platform(s)

Android

### Current Behavior

JFrog has [broken it's promise](https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/) to keep jcenter up "indefinitely".

> UPDATE 4/27/2021: We listened to the community and will keep JCenter as a read-only repository indefinitely. Our customers and the community can continue to rely on JCenter as a reliable mirror for Java packages.

They are currently [just redirecting to maven central](https://status.gradle.com/incidents/7x4wqd7zv715) which does not have anything before version 10

### Expected Behavior

Can build Capacitor v2 apps

### Code Reproduction

N/A

### Other Technical Details

N/A

### Additional Context

This issue is not for ionic-team/capacitor to do anything, but rather as a way for the community to find information when they search for this issue in their builds. Also I'd love if anyone has any workaround (other than upgrading, we are investigating that now). I have the .pom and .aar files for 7.0.0 but I'm unable to get them to work (by adding them to the source).

See also this same discussion on [apache/cordova-android](https://github.com/apache/cordova/discussions/414)

**Comment by jcesarmobile** [comment]
The incident has been resolved.

But I would encourage any users still using an old Capacitor version to migrate to Capacitor 5 and stop using jcenter

**Reply by tomchiverton** [reply]
How has this been resolved ? What do I need to do to update ?

**Reply by Fulvia1989** [reply]
The incident was one year ago, now the problem is that jCenter has been shut down definetly.
If you want to update to Capacitor 5, I guess you have to manually update the dependecies in your project (maybe following a step-by-step a guide from somewhere).
If you cannot update (like me)...I guess the only hope is to find  a copy of the .aar file and make it work somehow...

**Reply by tomchiverton** [reply]
I believe I was able to fix this by doing the following. It's at least past enough Google checks to reach our internal testers after a build today on https://dashboard.ionicframework.com/

We're only using some very basic integrations, so YMMV, and also the old build image `Linux - 2022.10`, as we're only doing this build to bump the target version before Google kick us out, and that's the version that last worked a year ago.

```
diff --git a/android/build.gradle b/android/build.gradle
index ad085c34555a1bd2220f39aa13c87c5ab568186d..6bb35717f6e8a3bb4a930500d895d7f2662bd8bf 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -4,7 +4,7 @@ buildscript {
     
     repositories {
         google()
-        jcenter()
+        mavenCentral()
     }
     dependencies {
         classpath 'com.android.tools.build:gradle:4.2.1'
@@ -20,7 +20,7 @@ apply from: "variables.gradle"
 allprojects {
     repositories {
         google()
-        jcenter()
+        mavenCentral()
     }
 }
 
diff --git a/android/variables.gradle b/android/variables.gradle
index 270dbb0e8c9d9cb2f713ce5e7c2c0e05fc5d0737..644753a5582816a64ef70d659f200210398270a4 100644
--- a/android/variables.gradle
+++ b/android/variables.gradle
@@ -10,5 +10,5 @@ ext {
     junitVersion = '4.13.1'
     androidxJunitVersion = '1.1.2'
     androidxEspressoCoreVersion = '3.3.0'
-    cordovaAndroidVersion = '7.0.0'
+    cordovaAndroidVersion = '10.1.1'
 }
```

**Comment by Fulvia1989** [comment]
My build failed today for the same reason.

**Reply by jcesarmobile** [reply]
JCenter was Sunset on August 15th, 2024
https://jfrog.com/blog/jcenter-sunset/

You should update to Capacitor 6, or at least to a version that doesn't use jcenter

**Reply by Fulvia1989** [reply]
Easier said than done. I cannot update Capacitor at the moment. Is there an alternative?

**Comment by sagadav** [comment]
I ran into this problem on a project with quasar js, and just changing the version didn't help me. It turned out that I use the second version of capacitor (in src-capacitor), so
```
cd src-capacitor
npm i -D @capacitor/cli@latest-5
npx cap migrate
```

Maybe this will help someone not to waste a lot of time.

**Comment by LukeGrave** [comment]
Hi @joshstrange and everyone else,

Is someone of you still in possession of the cordova framework 7.0.0 .pom and .aar files and is it possible to get a copy of it?

Thanks in advance,
Luke

**Reply by Fulvia1989** [reply]
I do have a copy of the .aar  file  that I managed to retrieve from an older project. 
I'm keeping it in one of my repos --> [cordova-framework-7](https://github.com/Fulvia1989/cordova-framework-7).

I don't have the .pom file thou, sorry..

Hope this helps anyway

**Comment by amethystdev1** [comment]
In case anyone is looking for a solution

- Download .pom, .aar from https://artifactory.appodeal.com/appodeal-public/org/apache/cordova/framework/7.0.0/ 
In case link above stops working, I made a backup at https://github.com/amethystdev1/apache-cordova-framework-v7/ 

- Run `mkdir -p ~/.m2/repository/org/apache/cordova/framework/7.0.0`
- Copy both `.aar` and `.pom` files to the `7.0.0` directory

- Create or modify `~/.gradle/init.gradle` to contain:
```gradle
allprojects {
    repositories {
        google()
        mavenCentral()
        maven {
            url uri("${System.properties['user.home']}/.m2/repository")
        }
    }
}
```

- Refresh Android studio project by clicking on the build error and you're ready to go.

**Reply by ford152** [reply]
thank you!

**Reply by muriloferes** [reply]
Thank you! You saved my life!

**Reply by UsmGH** [reply]
That saved my life

**Reply by LozanoEdd19** [reply]
This solution worked very well and gives us time to migrate the project to more recent versions. Thank you!!

**Reply by ViteRoberto** [reply]
Jun 2025 your answer still working. Works like a charm Thanks!

**Reply by jnettome** [reply]
thank you! working on july 25!

**Reply by HamzaAdam** [reply]
Can you pls tell how can i do it in my capacitor 3 app?
